payment/src/main/java/net/optile/payment/ui/widget/input/ExpiryDateInputMode.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019 optile GmbH
+ * https://www.optile.net
+ *
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more information.
+ */
+
+package net.optile.payment.ui.widget.input;
+
+import com.google.android.material.textfield.TextInputEditText;
+
+import android.text.InputFilter;
+import android.text.InputType;
+import android.text.method.DigitsKeyListener;
+
+/**
+ * InputMode for IBAN numbers
+ */
+public final class ExpiryDateInputMode extends EditTextInputMode {
+
+    public final static int MAXLENGTH = 7;
+    public final static String DIVIDER = " / ";
+    public final static String DATE_DIGITS = "0123456789/ ";
+
+    /**
+     * Construct an IBANInputMode
+     */
+    public ExpiryDateInputMode() {
+        super(MAXLENGTH, 0);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void apply(TextInputEditText editText) {
+        InputFilter[] filters = new InputFilter[1];
+        filters[0] = new InputFilter.LengthFilter(maxLength);
+        editText.setFilters(filters);
```

Comment:
Same number of lines but maybe a bit easier to read:
```suggestion
        editText.setFilters(new InputFilter[] {
            new InputFilter.LengthFilter(maxLength);
        });
```

Reply:
Changed this in the ExpiryDateInputMode and also in the other modes.